packages/lib-classifier/src/store/TutorialStore.spec.js

```diff
@@ -1,74 +1,337 @@
+let rootStore
 
-const get = () => {
-  return Promise.resolve({
-    body: {
-      tutorials: [tutorial]
-    }
-  })
-}
+const workflow = WorkflowFactory.build()
+
+const panoptesClient = stubPanoptesJs({ workflows: workflow })
 
 const medium = TutorialMediumFactory.build()
+const tutorial = TutorialFactory.build({ steps: [
+  { content: '# Hello', media: medium.id },
+  { content: '# Step 2' }
+] })
 
-const getAttachedImages = () => {
-  return Promise.resolve({
-    body: {
-      media: [medium]
-    }
-  })
+const clientStub = Object.assign({}, panoptesClient, {
+  tutorials: {
+    get: sinon.stub().callsFake(() => {
+      return Promise.resolve({
+        body: {
+          tutorials: [tutorial]
+        }
+      })
+    }),
+    getAttachedImages: sinon.stub().callsFake(() => {
+      return Promise.resolve({
+        body: {
+          media: [medium]
+        }
+      })
+    })
+  }
+})
+
+const authClientStubWithoutUser = {
+  checkCurrent: sinon.stub().callsFake(() => Promise.resolve(null)),
+  checkBearerToken: sinon.stub().callsFake(() => Promise.resolve(null))
 }
 
-const clientStub = { panoptes: { get }, tutorials: { getAttachedImages } }
+const authClientStubWithUser = {
+  checkCurrent: sinon.stub().callsFake(() => Promise.resolve(user)),
+  checkBearerToken: sinon.stub().callsFake(() => Promise.resolve(token))
+}
 
-const ROOT_STORE = types
-  .model('RootStore', {
-    tutorials: types.optional(TutorialStore, TutorialStore.create()),
-    workflows: types.frozen({})
+describe('Model > TutorialStore', function () {
+  afterEach(function () {
+    clientStub.tutorials.get.resetHistory()
+    clientStub.tutorials.getAttachedImages.resetHistory()
   })
-  .views(self => ({
-    get client () {
-      return getEnv(self).client
-    }
-  }))
 
-xdescribe('Model > TutorialStore', function () {
   it('should exist', function () {
     expect(TutorialStore).to.be.an('object')
   })
 
-  describe('when the observer observes that there is a workflow', function () {
-    let ROOT_STORE_INSTANCE = null
-    let WORKFLOW = null
-    before(function () {
-      WORKFLOW = WorkflowFactory.build()
+  it('should remain in an initialized state if there is no workflow', function () {
+    rootStore = RootStore.create({
+      tutorials: TutorialStore.create(),
+      workflows: WorkflowStore.create()
+    }, { authClient: authClientStubWithoutUser, client: clientStub })
 
-      ROOT_STORE_INSTANCE = ROOT_STORE.create({
-        workflows: {
-          active: WORKFLOW
-        }
-      }, { client: clientStub })
+    expect(rootStore.tutorials.loadingState).to.equal(asyncStates.initialized)
+  })
+
+  it('should set the tutorial if there is a workflow', function (done) {
+    rootStore = RootStore.create({
+      tutorials: TutorialStore.create(),
+      workflows: WorkflowStore.create()
+    }, { authClient: authClientStubWithUser, client: clientStub })
+    rootStore.workflows.setActive(workflow.id)
+      .then(() => {
```

Comment:
Minor point: Arrow functions are discouraged in mocha tests, but my understanding is that's only a big deal for tests that use `this`.

Reply:
I think it's an issue between expectations or describe blocks if you use arrow functions, but I think it's fine within a single expectation. The `it` function that is running this promise here is not using it.